applications/GeoMechanicsApplication/custom_elements/transient_thermal_element.h

```diff
@@ -74,6 +74,8 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) TransientThermalElement : public Ele
         if (GetGeometry().LocalSpaceDimension() == 1) {
             GetGeometry().DeterminantOfJacobian(det_J_container, this->GetIntegrationMethod());
             dN_dX_container = GetGeometry().ShapeFunctionsLocalGradients(this->GetIntegrationMethod());
+            std::transform(dN_dX_container.begin(), dN_dX_container.end(), det_J_container.begin(),
```

Comment:
It is not completely clear yet to me, why we need this division. Maybe it's a good idea to add a comment in the code?

Reply:
Because dN-dX has to be multiplied by inverse of Jacobian. In 1D inverse of jacobian is equal to inverse of determinant of Jacobian.